src/test/java/uk/gov/hmcts/reform/authorisation/validators/ServiceAuthTokenValidatorTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.authorisation.validators;
+
+import feign.FeignException;
+import feign.Response;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.reform.authorisation.ServiceAuthorisationApi;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.authorisation.exceptions.ServiceException;
+
+import java.util.Collections;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class ServiceAuthTokenValidatorTest {
+    private final ServiceAuthorisationApi api = mock(ServiceAuthorisationApi.class);
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void shouldValidateServiceAuthToken() {
+        final String serviceAuthToken = "service-auth-token";
+        final ServiceAuthTokenValidator validator = new ServiceAuthTokenValidator(api);
+
+        validator.validate(serviceAuthToken);
+        validator.validate(serviceAuthToken);
+
+        verify(api, times(2)).authorise(serviceAuthToken, new String[0]);
+    }
+
+    @Test
+    public void shouldThrowInvalidTokenException() {
```

Comment:
What do you think about about e.g. `shouldThrowInvalidTokenExceptionWhenServiceRespondsWithUnauthorised`?

Reply:
Really too long. Would be more reasonable if let's say tested some of the other 4xx response codes but then some copy paste happens. Unless extracting this test case to another parameterised test suite and prove code correctness fully. I actually fancy this idea while I was typing this message